### ISSUE TITLE
Segmentation fault at runtime

### ISSUE THREAD

**Issue by koaledu** [cryptocode/bio#15]
Followed the instructions and was able to build Bio but I get this when trying to run the REPL or run any code.

Tested on EndeavourOS with Zig 0.13.0 and Linux 6.10.7 for x86-64.
```
Segmentation fault at address 0x11eb9f0
/usr/lib/zig/compiler_rt/memcpy.zig:19:21: 0x11e905f in memcpy (compiler_rt)
            d[0] = s[0];
                    ^
/home/eduardo/bio/src/ast.zig:530:25: 0x106d23b in parse (bio)
                    atom.tok = tok;
                        ^
/home/eduardo/bio/src/ast.zig:446:25: 0x106e26f in parseMultipleExpressions (bio)
        try parser.parse();
                        ^
/home/eduardo/bio/src/ast.zig:437:40: 0x1097f5b in parseMultipleExpressionsFromReader__anon_7109 (bio)
        return parseMultipleExpressions(try source.toOwnedSlice(), file_path);
                                       ^
/home/eduardo/bio/src/intrinsics.zig:172:76: 0x10753e8 in import (bio)
        const expr_list = try ast.Parser.parseMultipleExpressionsFromReader(reader, try gc.allocator().dupe(u8, path));
                                                                           ^
/home/eduardo/bio/src/main.zig:25:34: 0x10744a1 in main (bio)
        _ = try intrinsics.import(interpreter, interpreter.env, args.items);
                                 ^
/usr/lib/zig/std/start.zig:524:37: 0x1075cce in main (bio)
            const result = root.main() catch |err| {
                                    ^
???:?:?: 0x70a2f10d4e07 in ??? (libc.so.6)
Unwind information for `libc.so.6:0x70a2f10d4e07` was not available, trace may be incomplete

???:?:?: 0x70a2f10d4ecb in ??? (libc.so.6)
???:?:?: 0x1058774 in ??? (???)
```

**Comment by cryptocode** [comment]
I'm unfortunately not able to reproduce this on my OS.

Are the examples working, for instance `zig-out/bin/bio run examples/triangles.lisp` ?  

Does a simple expression like `(+ 2 3) work in the REPL?

If those things crashes I'll try to set up an environment with EndeavourOS.

**Comment by koaledu** [comment]
No luck, I can't run the binary in any way.
![screenshot-2024-09-06-11-25-44](https://github.com/user-attachments/assets/5ff6ef24-c457-4fce-a700-74d36727cc5a)

**Comment by cryptocode** [comment]
Fixed by 54d80a6cd4ef8b5ac0df1ebb39d6da990e3c827a
